tests/unit_tests/test_invalid_build.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from magniv.build.build import build
+from tests.unit_tests.fixtures.test_files import (
+    INVALID_KWARG_FILE,
+    INVALID_TASK_KEY_FILE,
+    INVALID_TRIGGER_FILE,
+)
+
+# TODO: merge with TestBuild upon refactor there
+
+
+class TestBuildInvalid:
+    @pytest.fixture
+    def folder(self, tmpdir, file):
+        tmpdir.mkdir("tasks").join("main.py").write(file)
+        tmpdir.join("tasks/requirements.txt").write("magniv")
+        return str(tmpdir.join("tasks"))
+
+
+class TestBuildInvalidKwarg(TestBuildInvalid):
+    @pytest.fixture(scope="class")
+    def file(self):
+        return INVALID_KWARG_FILE
+
+    def test_invalid_build_raises_error(self, folder):
+        with pytest.raises(ValueError):
+            build(task_folder=folder)
```

Comment:
Can we abstract this boilerplate into base class?

```python
def build_folder():
  build(task_folder=folder)
def assert_build_error():
  with pytest.raises(ValueError):
     build(task_folder=folder)
```

or something like that... make it easy. the `assert_build_error` might be less trivial not 100% that would work

Reply:
I tried something like this but it gets weird because if you put it in the base class I don't think it gets invoked correctly for each of the subclasses. Like because it's repeated currently it gets run as an individual test for each subclass, but if you put it in the base it just gets confused.

Definitely a "code smell" to have this boilerplate repeated so I think there's probably a better way, but I haven't been able to figure it out.

Reply:
Yeah just tried again and it doesn't work but I was wrong about why - if you put it in the base class it doesn't sequence right with the definition of `file`. It tries to run the test in the base class using the stub, and also fails in the subclass invocation (not really sure why on that one)

I am inclined to leave this alone for the time being, but maybe something will come along when we refactor `test_build`